src/front/glsl/variables.rs

```diff
@@ -465,8 +465,8 @@ impl Parser {
             StorageQualifier::AddressSpace(mut space) => {
                 match space {
                     AddressSpace::Storage { ref mut access } => {
-                        if let Some((restricted_access, _)) = qualifiers.storage_acess.take() {
-                            access.remove(restricted_access);
+                        if let Some((allowed_access, _)) = qualifiers.storage_acess.take() {
+                            *access = allowed_access;
```

Comment:
shouldn't this be `*access &= allowed_access;` instead?

Reply:
It's the same thing since the only way to change access which defaults to `all` is trough memory qualifiers